static/scripts/PlotHandler.js

```diff
@@ -0,0 +1,121 @@
+import { addNewPlotEntry } from "./workspace";
+
+class PlotHandler {
+    constructor() {
+        this.plotMap = new Map();
+    }
+    //the data forwarded contains the information: data(yValue,datasetName,plotName,plotType)
+    //forward the data to the correct PlotWorker or create one if necessary 
+    updateValue(data) {
+        let plotName = data[3];
+        let requestedPlot = this.plotMap.get(plotName);
+        if (requestedPlot == undefined) {
+            requestedPlot = new PlotWorker(plotName, data[4]);
+            this.plotMap.set(plotName, requestedPlot);
+        }
+        requestedPlot.updateValue(data);
+    }
+
+    drawPlots() {
+        this.plotMap.forEach((value) => value.drawPlot());
+    }
+
+    clearPlots() {
+        this.plotMap = new Map();
+    }
+}
+
+class PlotWorker {
+    constructor(name, plotType) {
+        this.plotName = name;
+        this.plotData = new Map();
+        this.plotType = plotType;
+        let canvasNumber = addNewPlotEntry(name);
+        let canvasID = 'plot-' + canvasNumber + '-canvas';
+        this.chartArea = document.getElementById(canvasID).getContext('2d');
+        this.myChart = null;
+        this.chartExists = false;
+        this.labels = [];
+        this.iteration = 0;
+    }
+
+    //collect data during runtime 
+    updateValue(data) {
+        let datasetName = data[2];
+        if (this.iteration > 0) {
+            // new runs produce new datasets which can be distinguished by their ending 
+            datasetName += "_" + this.iteration;
+        }
+        let dataset = this.plotData.get(datasetName);
+        if (dataset == undefined) {
```

Comment:
small hint: you could just use `if("dataset)` instead of checking equality with `undefined`